hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/NetworkGetByKeyHandler.java

```diff
@@ -57,13 +61,14 @@ public Response createEmptyResponse(@NonNull final ResponseHeader header) {
     @Override
     public void validate(@NonNull final QueryContext context) throws PreCheckException {
         requireNonNull(context);
-        throw new UnsupportedOperationException("Not implemented");
+        throw new PreCheckException(NOT_SUPPORTED);
     }
 
     @Override
     public Response findResponse(@NonNull final QueryContext context, @NonNull final ResponseHeader header) {
+        // this code should never be executed, as validate() should fail before we get here
         requireNonNull(context);
         requireNonNull(header);
-        throw new UnsupportedOperationException("Not implemented");
+        throw new UnsupportedOperationException("Not supported");
```

Comment:
why not to use the same ENUM

Reply:
Here in findResponse we can't throw a PreCheckException, since the superclass doesn't throw it. Instead we throw UnsupportedOperationException which takes a String as an argument.
Really it doesn't much matter what we do here since the transaction should fail on validate and never reach findResponse.